File: tests/dynamic_loading_test.py
import pytest
from pages import dynamic_loading_page


@pytest.fixture
def dynamic_loading(driver):
    return dynamic_loading_page.DynamicLoadingPage(driver)

@pytest.mark.deep
def test_hidden_element(dynamic_loading):
    dynamic_loading.load_example("1")
    assert dynamic_loading.finish_text_present()

@pytest.mark.deep
def test_rendered_element(dynamic_loading):
    dynamic_loading.load_example("2")
    assert dynamic_loading.finish_text_present()

File: tests/DynamicLoadingTest.js
'use strict';
var assert = require('assert')
var webdriver = require('selenium-webdriver');
var test = require('selenium-webdriver/testing');
var DriverFactory = require('../lib/DriverFactory');
var DynamicLoadingPage = require('../pages/DynamicLoadingPage');

test.describe('Dynamic Loading', function() {
  this.timeout(30000); // for mocha
  var dynamicLoading;

  test.beforeEach(function() {
    dynamicLoading = new DynamicLoadingPage(DriverFactory.build());
  });

  test.afterEach(function() {
    DriverFactory.quit();
  });

  test.it('hidden element', function() {
    dynamicLoading.loadExample('1');
    dynamicLoading.finishTextPresent().then(function(elementDisplayed) {
      assert.equal(elementDisplayed, true, 'Finish text not displayed');
    });
  });

  test.it('rendered element', function() {
    dynamicLoading.loadExample('2');
    dynamicLoading.finishTextPresent().then(function(elementDisplayed) {
      assert.equal(elementDisplayed, true, 'Finish text not displayed');
    });
  });
});
